gqcp/tests/Operator/SecondQuantized/USQTwoElectronOperator_test.cpp

```diff
@@ -0,0 +1,452 @@
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+// 
+// GQCG-gqcp is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with GQCG-gqcp.  If not, see <http://www.gnu.org/licenses/>.
+// 
+#define BOOST_TEST_MODULE "USQTwoElectronOperator"
+
+#include <boost/test/unit_test.hpp>
+
+#include "Operator/SecondQuantized/USQTwoElectronOperator.hpp"
+
+#include "Utilities/linalg.hpp"
+#include "Utilities/miscellaneous.hpp"
+
+#include <boost/math/constants/constants.hpp>
+
+
+/**
+ *  Check the interface for constructing SQTwoElectronOperators from Tensors
+ */
+BOOST_AUTO_TEST_CASE ( USQTwoElectronOperator_constructor ) {
+
+    // Check a correct constructor
+    const GQCP::QCRankFourTensor<double> tensor (3);
+    GQCP::ScalarUSQTwoElectronOperator<double> O (tensor, tensor, tensor, tensor);
+
+
+    // Check a faulty constructor
+    GQCP::Tensor<double, 4> tensor2 (3, 3, 3, 2);
+    BOOST_CHECK_THROW(GQCP::ScalarUSQTwoElectronOperator<double> O2 (tensor2, tensor2, tensor2, tensor2), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::ScalarUSQTwoElectronOperator<double> O2 (tensor, tensor2, tensor, tensor), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::ScalarUSQTwoElectronOperator<double> O2 (tensor2, tensor, tensor2, tensor), std::invalid_argument);
+    BOOST_CHECK_THROW(GQCP::ScalarUSQTwoElectronOperator<double> O2 (tensor2, tensor2, tensor2, tensor), std::invalid_argument);
+
+}
+
+
+/**
+ *  Check if the zero constructor really sets is parameters to all zeros
+ */
+BOOST_AUTO_TEST_CASE ( USQTwoElectronOperator_zero_constructor ) {
+
+    const size_t K = 2;
+    GQCP::ScalarUSQTwoElectronOperator<double> op {K}; // should initialize zero's
+
+    // Create a reference zero tensor
+    GQCP::QCRankFourTensor<double> ref (K);
+
+    for (size_t i = 0; i < K; i++) {
+        for (size_t j = 0; j < K; j++) {
+            for (size_t k = 0; k < K; k++) {
+                for (size_t l = 0; l < K; l++) {
+                    ref(i,j,k,l) = 0;
+                }
+            }
+        }
+    }
+
+    BOOST_CHECK_EQUAL(op.alphaAlphaDimension(), K);
+    BOOST_CHECK_EQUAL(op.alphaBetaDimension(), K);
+    BOOST_CHECK_EQUAL(op.betaAlphaDimension(), K);
+    BOOST_CHECK_EQUAL(op.betaBetaDimension(), K);
+
+    BOOST_CHECK(op.alphaAlphaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.alphaBetaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.betaAlphaParameters().isApprox(ref, 1.0e-08));
+    BOOST_CHECK(op.betaBetaParameters().isApprox(ref, 1.0e-08));
+
```

Comment:
```suggestion
    BOOST_CHECK(op.alphaAlphaParameters().isZero(1.0e-08));
    BOOST_CHECK(op.alphaBetaParameters().isZero(1.0e-08));
    BOOST_CHECK(op.betaAlphaParameters().isZero(1.0e-08));
    BOOST_CHECK(op.betaBetaParameters().isZero(1.0e-08));
```

`Eigen` provides a simpler `API` to check if a `Tensor` is actually zero.

Reply:
I just realised, after committing of course, that this won't work. The QCRankFourTensor doesn't have an "isZero()". Should I change it back to the way it was, or is there a better/more efficient way to do this?